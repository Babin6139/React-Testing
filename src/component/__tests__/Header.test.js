
import React from "react";
import Header from "../Header";
import {render} from "@testing-library/react";

describe("Tests for Header Component",()=>{
    
    it("snapshot header component",()=>{
        const mockText="This is the string for testing";
        const {asFragment}=render(<Header text={mockText}/>);
        expect(asFragment(<Header text={mockText}/>)).toMatchSnapshot();
    });

    it("should render header component with right text",()=>{
        const mockText="This is the string for testing";
        const {getByText}=render(<Header text={mockText}/>);
        expect(getByText(mockText)).not.toBeNull();
    });

});