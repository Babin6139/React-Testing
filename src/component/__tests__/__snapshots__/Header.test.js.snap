// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Header Component snapshot header component 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    This is the string for testing
  </div>
</DocumentFragment>
`;
